src/features/RenameSymbol.ts

```diff
@@ -0,0 +1,132 @@
+    message : string
+}
+
+export const RenameSymbolRequestType = new RequestType<IRenameSymbolRequestArguments, IRenameSymbolRequestResponse, void>("powerShell/renameSymbol");
+export const PrepareRenameSymbolRequestType = new RequestType<IPrepareRenameSymbolRequestArguments, IPrepareRenameSymbolRequestResponse, void>("powerShell/PrepareRenameSymbol");
+
+export class RenameSymbolFeature extends LanguageClientConsumer implements RenameProvider {
+    public override onLanguageClientSet(_languageClient: LanguageClient): void {
+        throw new Error("Method not implemented.");
+    }
+    private command: vscode.Disposable;
+
+    constructor() {
+        super();
+        this.command = vscode.commands.registerCommand("PowerShell.RenameSymbol", () => {
+            throw new Error("Not implemented");
+
+        });
+    }
+    public dispose() :void{
+        this.command.dispose();
+    }
+    public async provideRenameEdits(document: TextDocument, position: Position, newName: string, _token: CancellationToken): Promise<WorkspaceEdit | undefined> {
+
+        const req:IRenameSymbolRequestArguments = {
+            FileName : document.fileName,
+            Line: position.line,
+            Column : position.character,
+            RenameTo : newName,
+        };
+
+        try {
+            const client = await LanguageClientConsumer.getLanguageClient();
+            const response = await client.sendRequest(RenameSymbolRequestType, req);
+
+            if (!response.changes.length) {
+                return undefined;
+            }
+
+            const edit = new WorkspaceEdit();
+            response.changes.forEach(change => {
+                const uri = Uri.file(change.fileName);
+
+                change.changes.forEach(change => {
+                    edit.replace(uri,
+                        new Range(change.startLine, change.startColumn, change.endLine, change.endColumn),
+                        change.newText);
+                });
+            });
+            return edit;
+        }catch (error) {
+            return undefined;
+        }
+    }
+    public async prepareRename(document: vscode.TextDocument, position: vscode.Position, _token: vscode.CancellationToken): Promise<vscode.Range | {
+        range: vscode.Range; placeholder: string;} | null> {
+
+        const req:IRenameSymbolRequestArguments = {
+            FileName : document.fileName,
+            Line: position.line,
+            Column : position.character,
+        };
+
+        try {
+            const client = await LanguageClientConsumer.getLanguageClient();
+            const response = await client.sendRequest(PrepareRenameSymbolRequestType, req);
+
+            if (!response.message) {
+                return null;
+            }
+            const wordRange = document.getWordRangeAtPosition(position);
+            if (!wordRange) {
+                return Promise.reject("Not a valid location for renaming.");
```

Comment:
Nit: Since this is an async function, why not just throw? should have the same result and is more idiomatic.

Reply:
Fixed, thanks!